core/state/state_object.go

```diff
@@ -213,6 +213,9 @@ func (s *stateObject) GetCommittedState(key common.Hash) common.Hash {
 			}
 			value.SetBytes(content)
 		}
+		if s.db.collectWitness && s.db.prefetcher != nil && s.data.Root != types.EmptyRootHash {
+			s.db.prefetcher.prefetch(s.addrHash, s.origin.Root, s.address, [][]byte{key[:]})
+		}
```

Comment:
Shouldn't this be moved out from the if-clause? As in, do it regardless of whether we're in snapshot mode or trie mode